api/src/classes/company.py

```diff
@@ -28,19 +29,15 @@ def __init__(self, session: Session, company: Company) -> None:
         self.company: Company = company
 
     @classmethod
-    def fetch_company(
-        cls, session: Session, *, name: str | None = None, owner_id: int | None = None, company_id: int | None = None
-    ) -> Self:
+    def fetch_company(cls, session: Session, company_id: str) -> Self:
         """Return instance with target Company, if it exists.
 
         :param company_id: ID of the Company to search for.
         :param session: Database session.
-        :param name: Name of the company.
-        :param owner_id: ID of the Owner.
         :return: Instance with the fetched company.
         """
         fetched_company: Company | None = session.exec(
-            select(Company).where(or_(Company.id == company_id, Company.name == name, Company.owner_id == owner_id))
+            select(Company).where(Company.owner_id == company_id, not_(Company.is_bankrupt))
```

Comment:
Possibly try to find the latest company if nothing match this query (i.e. all company the user own is bankrupt)

Reply:
Other functions rely on this one to get the user's active company for updates, deleting, purchasing etc.

I could possibly add an endpoint to return all the companies that a user has had? Like `/user/{id}/companies`?

Reply:
Maybe, that's just a small thought and adding an additional endpoint would be unlikely to be used.
The only reason I could think about is getting the company why after they go bankrupt to tell the user (but maybe that could be done just before caching)

So... Just ignore this :)